### ISSUE TITLE
PySpark as another distributed dataframe

### ISSUE THREAD

**Issue by Lightslayer** [modin-project/modin#1117]
I recently discovered modin and loved the clean approach to working with large dataframes in a simple manner. One of the things that struck me was that the [Modin architecture](https://modin.readthedocs.io/en/latest/architecture.html 
) appears to be highly extensible so that developers can, in principle, create their own distributed dataframes/backends beyond the existing Pandas + Ray/Dask implementations.

As such I wonder how cool a feature it would be to be able to create Modin adaptors to PySpark and be able to leverage the power of Spark, using the more familiar Modin/Pandas API approach?

It is probably a huge amount of work, but would mean that Modin is the go-to package for all your parallel/distributed compute dataframe needs :-)

**Comment by devin-petersohn** [comment]
Hi @Lightslayer, thanks for the question!

We have considered it, but Spark unfortunately doesn't have all of the properties we would need to implement the pandas API. I think we could use our architecture to coerce some properties of Spark into what they would need to be, but that would take a significant effort, and with limited engineering resources, we wouldn't be able to get to it soon. Koalas (https://github.com/databricks/koalas) is an effort to do this, but they've been clear from the beginning that they can't support everything in pandas due to the limitations and strict nature of Spark. In Modin, supporting the full pandas API has been a design choice from the start.

Dataframes is the focus of my PhD, and we recently uploaded a preprint about dataframes in general: https://arxiv.org/pdf/2001.00888.pdf. TL;DR Spark is not really a dataframe because it doesn't support everything that you can do in pandas/R/S dataframes and deviates in some very significant ways.

**Comment by Hoeze** [comment]
Maybe it would already be enough providing easy conversion mechanisms?
E.g. `pd.from_spark()` that reads/computes partitions on-demand?

**Comment by Hoeze** [comment]
@devin-petersohn Maybe it would be enough to provide simple conversion mechanisms inside Modin?
E.g. some `mpd.from_spark()/mpd.to_spark()` that reads/computes partitions on-demand?

**Comment by danilopeixoto** [comment]
I found this interesting and cool. Is this new [Pandas API on Spark](https://spark.apache.org/docs/latest/api/python/getting_started/quickstart_ps.html ) still far from the real Pandas API?

**Comment by devin-petersohn** [comment]
@danilopeixoto Yes, that is quite an interesting effort. Collaboration with those folks is a possibility in the future! Their implementation could benefit from our expertise in this area, considering some of the design decisions are more Spark-inspired than pandas-inspired.
